apps/openmw/mwrender/animation.cpp

```diff
@@ -113,19 +120,89 @@ namespace
             texEnv->setOperand2_RGB(osg::TexEnvCombine::SRC_COLOR);
 
             stateset->setTextureAttributeAndModes(mTexUnit, texEnv, osg::StateAttribute::ON);
+
+            // Reduce the texture list back down by one when a temporary glow finishes
+            // to allow FindLowestUnusedTexUnitVisitor to choose the same texunit again.
+            if (mDone)
+                stateset->getTextureAttributeList().resize(stateset->getTextureAttributeList().size() - 1);
         }
 
         virtual void apply(osg::StateSet *stateset, osg::NodeVisitor *nv)
         {
+            if (mDone)
+                return;
+
+            // If there is a permanent enchantment glow already on this object, give the
+            // permanent glow a pointer to a temporary glow added in a nested update callback
+            // The permanent glow will remove the temporary glow when it finishes, allowing
+            // the permanent glow to display its glow texture cycling properly.
+            if (mWatchedSpellGlow != NULL && mWatchedSpellGlow->isDone())
+            {
+                mNode->removeUpdateCallback(mWatchedSpellGlow);
+                mWatchedSpellGlow = NULL;
+            }
```

Comment:
That is an ugly workaround. If you want to use multiple updaters then the proper way to do it would be using the ```CompositeStateSetUpdater```.
```
osg::ref_ptr<CompositeStateSetUpdater> composite (new CompositeStateSetUpdater);
composite->addController( spellglowupdater1 );
composite->addController( spellglowupdater2 );
mObjectRoot->addUpdateCallback(composite);
```

That way both updaters are not messing with the other's StateSet.

However... there is also a problem with the feature you are trying to implement. You want to have a second glow texture applied on top of the first one. Correct? Problem is that the shader isn't set up to handle that. It'll work in fixed function mode, but the shader would only show one of the glow textures. Extending the shader code to support an arbitrary number of glow textures would be pretty complicated.

Easiest thing would be not to allow spell glow on things that already glow but I guess you're too keen on that feature ;)

I suggest to just temporarily override the color of the first glow updater for the duration of the spell glow. Should look close enough.